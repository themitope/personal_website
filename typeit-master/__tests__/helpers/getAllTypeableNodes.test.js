import getAllTypeableNodes from "../../src/helpers/getAllTypeableNodes";

test("Should return the single text node.", () => {
  setHTML`
    <div id="element">
      This is text.
    </div>
  `;

  let el = document.getElementById("element");
  let result = getAllTypeableNodes(el);

  expect(result).toMatchSnapshot();
});

test("Should return multiple text nodes.", () => {
  setHTML`
    <div id="element">
      This is text. <span>And here is more.</span>
    </div>
  `;

  let el = document.getElementById("element");
  let result = getAllTypeableNodes(el);

  expect(result).toMatchSnapshot();
});

test("Should return break tag as separate node.", () => {
  setHTML`
    <div id="element">
      This text will break. <br> It broke.
    </div>
  `;

  let el = document.getElementById("element");
  let result = getAllTypeableNodes(el);

  expect(result).toMatchSnapshot();
});

test("Should return break tags as when they bookend string.", () => {
  setHTML`
    <div id="element">
      <br>Text in the middle</br>
    </div>
  `;

  let el = document.getElementById("element");
  let result = getAllTypeableNodes(el);

  expect(result).toMatchSnapshot();
});

test("Should return correct text nodes when nested deeply.", () => {
  setHTML`
    <div id="element">
      Text #1. <span>Text #2. <em>Text #3.</em>
    </div>
  `;

  let el = document.getElementById("element");
  let result = getAllTypeableNodes(el);

  expect(result).toMatchSnapshot();
});

test("Should not return nodes that live inside ignored node.", () => {
  setHTML`
    <div id="element"><span id="cursor">|<i>X</i></span></div>
  `;

  let el = document.getElementById("element");
  let elToIgnore = document.getElementById("cursor");

  // Add separate text nodes.
  el.insertBefore(document.createTextNode("a"), elToIgnore);
  el.insertBefore(document.createTextNode("b"), elToIgnore);
  el.insertBefore(document.createTextNode("c"), elToIgnore);

  let result = getAllTypeableNodes(el, elToIgnore);

  expect(result).toMatchSnapshot();
});
