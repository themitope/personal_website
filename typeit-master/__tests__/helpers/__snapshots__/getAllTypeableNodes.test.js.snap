// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not return nodes that live inside ignored node. 1`] = `
Array [
  a,
  b,
  c,
]
`;

exports[`Should return break tag as separate node. 1`] = `
Array [
        This text will break. ,
  <br />,
   It broke.    ,
]
`;

exports[`Should return break tags as when they bookend string. 1`] = `
Array [
  <br />,
  Text in the middle,
  <br />,
]
`;

exports[`Should return correct text nodes when nested deeply. 1`] = `
Array [
        Text #1. ,
  Text #2. ,
  Text #3.,
]
`;

exports[`Should return multiple text nodes. 1`] = `
Array [
        This is text. ,
  And here is more.,
]
`;

exports[`Should return the single text node. 1`] = `
Array [
        This is text.    ,
]
`;
